drivers/node/ssh/ssh.go

```diff
@@ -138,11 +160,11 @@ func (s *ssh) ShutdownNode(n node.Node, options node.ShutdownNodeOpts) error {
 		shutdownCmd = "halt"
 	}
 
-	t := func() error {
+	t := func() (interface{}, error) {
```

Comment:
These `t` functions don't have to have return type as interface{}. Only the generic `DoRetryWithTimeout` task package function should have generic return types. `t` is a user function that knows it's returning type is a string. So `t` should have a return type `(string, error)`

Do the same for other places in the review.

Reply:
I think that should be done on a case to case basis because currently none of the user tasks t() are using the string part yet.

Reply:
An `interface{}` should be used when you defined a generic API or struct that users will consume and you don't know what data type you might want to use. `t` here is a function variable, neither an API nor a generic data structure. API's like `DoRetryWithTimeout` are good candidates of using `interface{}` as return type since it provides a generic functionality and doesn't make assumptions about what callers might want.

In 
```
 t := func() (interface{}, error) {
      return s.doCmd(addr, shutdownCmd, true)
} 
```

You exactly know the return type since  `s.doCmd(...)` returns the string type. Infact, any other return type when defining `t` won't work.

While the code does compile, one shouldn't use `interface{}` as a placeholder for all types when we exactly know the returning type.